### ISSUE TITLE
Allow node prop-types in SideNavMenu & TooltipDefinition

### ISSUE THREAD

**Issue by kubijo** [carbon-design-system/carbon#7146]
Would you see any problem with allowing `React.ReacNode` (in typescript sense) to be passed into these?
- https://github.com/carbon-design-system/carbon/blob/master/packages/react/src/components/UIShell/SideNavMenu.js#L69
- https://github.com/carbon-design-system/carbon/blob/master/packages/react/src/components/TooltipDefinition/TooltipDefinition.js#L104

With the `SideNavMenu` it's about our design team wanting to put little charts into menu items & `TooltipDefinition` change is about being able to render icon as the "trigger text". Both of these work for us just fine. It's just about the annoying development mode warnings caused by `prop-types`.

## Detailed description

> What version of the Carbon Design System are you using?

`carbon-components-react@7.22.0`

**Comment by tw15egan** [comment]
@kubijo For the first part, you should be able to use the `renderIcon` prop to pass down an icon you want to be rendered on the menu.

For the second part, it seems like this is unlikely to be added due to a11y reasons:

> I think moving forward we'll want to keep the use-case for TooltipDefinition pretty constrained whereit's a tooltip for a word in a paragraph and the tooltip itself is text-only.

https://github.com/carbon-design-system/carbon/pull/6475#issuecomment-673509866

You could just the [normal icon tooltip](https://react.carbondesignsystem.com/?path=/story/tooltip--only-custom-icon) to achieve this though.

**Comment by kubijo** [comment]
As far as I know the `TooltipDefinition` is the only component that had desired behavior in that it's triggered by hover & works on mobile… but do correct me please if I'm mistaken on that and it can be achieved by a different component while also being able to give it any children component.

The `renderIcon`, I must concur, might work with style hacking (we need those charts on the other side) - I'll try that ;-)

**Comment by kubijo** [comment]
Anyway to ilustrate our usecase, which I find to be in line with what you have in mind…
![TooltipDefinition](https://user-images.githubusercontent.com/11244314/97494217-4e8aa600-1966-11eb-9c37-0b393cab7df2.png)

Here you can see that the content we are "annotating" is essentially a textual one, but it's a component nonetheless as it needs some formatting & context access… so while I get your reasoning I find it more limiting then neccessary and I'd let people do what they need with the burden of correctness being on them (which they'll do anyway, be it with some annoying errors)

**Comment by tay1orjones** [comment]
In v11 we introduced Popover, as well as new implementations of Tooltip, Toggletip, and TooltipDefinition that all use Popover under the hood. This mixture of components should let you achieve all the use cases outlined here.

I'm going to close this one out as I believe it's resolved, but let us know if there's anything left to tackle here.
